compiler/executable/pipeline.rs

```diff
@@ -171,14 +175,31 @@ pub(crate) fn compile_pipeline_stages(
     functions: &ExecutableFunctionRegistry,
     annotated_stages: Vec<AnnotatedStage>,
     input_variables: impl Iterator<Item = Variable>,
+    selected_variables: &Vec<Variable>,
 ) -> Result<(HashMap<Variable, VariablePosition>, Vec<ExecutableStage>), ExecutableCompilationError> {
     let mut executable_stages: Vec<ExecutableStage> = Vec::with_capacity(annotated_stages.len());
     let input_variable_positions =
         input_variables.enumerate().map(|(i, var)| (var, VariablePosition::new(i as u32))).collect();
+
     for stage in annotated_stages {
+        let selected_variables = selected_variables.iter().cloned().chain(stage.referenced_variables()).collect();
```

Comment:
On a new stage, we need not only the final selected variable, but also the variables our stages require, so we add it here.